selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java

```diff
@@ -0,0 +1,128 @@
+import static org.eclipse.che.selenium.pageobject.CodenvyEditor.MarkerLocator.ERROR_OVERVIEW;
+import static org.openqa.selenium.Keys.*;
+import static org.openqa.selenium.Keys.SPACE;
+import static org.testng.Assert.assertEquals;
+
+import com.google.inject.Inject;
+import java.net.URL;
+import java.nio.file.Paths;
+import org.eclipse.che.commons.lang.NameGenerator;
+import org.eclipse.che.selenium.core.client.TestProjectServiceClient;
+import org.eclipse.che.selenium.core.workspace.InjectTestWorkspace;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.pageobject.AskForValueDialog;
+import org.eclipse.che.selenium.pageobject.CodenvyEditor;
+import org.eclipse.che.selenium.pageobject.Consoles;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.Wizard;
+import org.eclipse.che.selenium.pageobject.intelligent.CommandsPalette;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/** @author Musienko Maxim */
+public class TypeScriptEditingTest {
+  private static final String PROJECT_NAME =
+      NameGenerator.generate(TypeScriptEditingTest.class.getSimpleName(), 4);
+  private static final String PATH_TO_GREETER_FILE = PROJECT_NAME + "/Greeter.ts";
+  private static final String PATH_TO_PRINT_TEST_FILE = PROJECT_NAME + "/printTest.ts";
+
+  @InjectTestWorkspace(template = ECLIPSE_NODEJS)
+  private TestWorkspace workspace;
+
+  @Inject private Ide ide;
+  @Inject private Menu menu;
+  @Inject private Wizard wizard;
+  @Inject private Consoles consoles;
+  @Inject private CodenvyEditor editor;
+  @Inject private CommandsPalette commandsPalette;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private AskForValueDialog askForValueDialog;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    URL resource = getClass().getResource("/projects/type-script-simple-project");
+    testProjectServiceClient.importProject(
+        workspace.getId(), Paths.get(resource.toURI()), PROJECT_NAME, NODE_JS);
+    ide.open(workspace);
+
+    projectExplorer.waitVisibleItem(PROJECT_NAME);
+    projectExplorer.quickExpandWithJavaScript();
+    projectExplorer.openItemByPath(PATH_TO_GREETER_FILE);
+  }
+
+  @Test
+  public void checkMainFeaturesTypeScriptLS() {
+    String intitTypeScriptLanguageServerMessage =
+        String.format(
+            "Finished language servers initialization, file path '/%s'", PATH_TO_GREETER_FILE);
+
+    consoles.waitExpectedTextIntoConsole(intitTypeScriptLanguageServerMessage);
+    checkCodeValidation();
+    checkCodeAssistant();
+    checkGoToDefinition();
+  }
+
+  private void checkCodeValidation() {
+    final int expectedValueOfErrorMarkers = 9;
```

Comment:
from the readability point it's better to divide tverification by given/when/then sections.

Reply:
Maybe "expectedCountOfErrorMarkers" better?